packages/data-context/src/polling/poller.ts

```diff
@@ -4,35 +4,50 @@ import debugLib from 'debug'
 
 const debug = debugLib('cypress:data-context:polling:Poller')
 
-export class Poller<E extends EventType> {
+export class Poller<E extends EventType, M> {
   constructor (private ctx: DataContext,
     private event: E,
     private pollingInterval: number,
     private callback: () => Promise<any>) {}
 
   #timeout?: NodeJS.Timeout
 
+  #subscriptionCount: number = 0
+  #subscriptions: Record<string, { meta: M | undefined }> = {}
```

Comment:
Should we just use a `#subscriptions: Map<string, { meta: M | undefined }> = new Map()` and instead of having two variables, one for the data and one for the count, just do `subscriptions.size()` to see the count? Then we do not need to track that size separately - single source of truth.

Reply:
I did it this way in order to create an auto incrementing "id" as the key for each map entry.  I will rename the variable to `subscriptionId` to make that more clear.